obspy/io/gcf/core.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+"""
+GCF bindings to ObsPy core module.
+"""
+
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+
+from obspy import Stream, Trace, UTCDateTime
+
+from . import libgcf
+
+
+def merge_gcf_stream(st):
+    """
+    Merges GCF stream (replacing Stream.merge(-1) for headonly=True)
+
+    :type st: :class:`~obspy.core.stream.Stream`
+    :param st: GCF Stream object whith no data
+    :rtype: :class:`~obspy.core.stream.Stream`
+    :returns: Stream object containing header and data.
+    """
+    traces = []
+    for tr in st:
+        merged = 0
+        delta = tr.stats.delta
+        starttime = tr.stats.starttime
+        endtime = tr.stats.endtime
+        for trace in traces:
+            if tr.id == trace.id and delta == trace.stats.delta \
+               and not starttime == trace.stats.starttime:
+                if 0 < starttime - trace.stats.endtime <= delta:
+                    trace.stats.npts += tr.stats.npts
+                    merged = 1
+                    break
+                elif 0 < trace.stats.starttime - endtime <= delta:
+                    trace.stats.starttime = UTCDateTime(starttime)
+                    trace.stats.npts += tr.stats.npts
+                    merged = 1
+                    break
+        if not merged:
+            traces.append(tr)
```

Comment:
You could get rid of the `merged` flag using a `for/else` statement.. I don't care if you change it here, it can also stay as is.
But maybe you haven't seen that construct yet, and it is kind of slick, especially for search loops that raise an Exception in the else clause if the expected item is not found, so FYI (also see e.g. [this post](http://nedbatchelder.com/blog/201110/forelse.html)):

```python
    traces = []
    for tr in st:
        delta = tr.stats.delta
        starttime = tr.stats.starttime
        endtime = tr.stats.endtime
        for trace in traces:
            if tr.id == trace.id and delta == trace.stats.delta \
               and not starttime == trace.stats.starttime:
                if 0 < starttime - trace.stats.endtime <= delta:
                    trace.stats.npts += tr.stats.npts
                    break
                elif 0 < trace.stats.starttime - endtime <= delta:
                    trace.stats.starttime = UTCDateTime(starttime)
                    trace.stats.npts += tr.stats.npts
                    break
        else:
            traces.append(tr) 
```